hermes/test/test_parsers.py

```diff
@@ -0,0 +1,271 @@
+        event_id = 'S112233'
+        with self.assertRaises(NonLocalizedEvent.DoesNotExist):
+            NonLocalizedEvent.objects.get(event_id=event_id)
+        message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=get_lvc_notice_text(type='LVC_PRELIMINARY', event_id=event_id)
+        )
+        self.assertTrue(GCNLVCNoticeParser().parse(message))
+        event = NonLocalizedEvent.objects.get(event_id=event_id)
+        self.assertEqual(event.event_id, event_id)
+
+    def test_nonlocalizedevent_sequences_created(self):
+        event_id = 'S112233'
+        with self.assertRaises(NonLocalizedEvent.DoesNotExist):
+            NonLocalizedEvent.objects.get(event_id=event_id)
+        message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=get_lvc_notice_text(type='LVC_PRELIMINARY', event_id=event_id, sequence_number=1)
+        )
+        self.assertTrue(GCNLVCNoticeParser().parse(message))
+        message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=get_lvc_notice_text(type='LVC_INITIAL', event_id=event_id, sequence_number=2)
+        )
+        self.assertTrue(GCNLVCNoticeParser().parse(message))
+        # Add a duplicate of one sequence_number to show it does not get added
+        message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=get_lvc_notice_text(type='LVC_INITIAL', event_id=event_id, sequence_number=2)
+        )
+        self.assertTrue(GCNLVCNoticeParser().parse(message))
+        sequences = NonLocalizedEventSequence.objects.filter(event__event_id=event_id)
+        self.assertEqual(sequences.count(), 2)
+        self.assertEqual(sequences[0].sequence_number, 1)
+        self.assertEqual(sequences[0].sequence_type, 'PRELIMINARY')
+        self.assertEqual(sequences[1].sequence_number, 2)
+        self.assertEqual(sequences[1].sequence_type, 'INITIAL')
+
+    def test_fail_to_parse_if_title_doesnt_contain_keywords(self):
+        # Expected keywords are LVC, GCN, and NOTICE
+        bad_message = 'TITLE:            BAD NOTICE\nTRIGGER_NUM:       S112233\nSEQUENCE_NUM:      1'
+        message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=bad_message
+        )
+        self.assertFalse(GCNLVCNoticeParser().parse(message))
+        message.refresh_from_db()
+        self.assertIsNone(message.data)
+        self.assertEqual(message.title, '')
+
+
+class TestLVCCounterpartParser(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.event_id = 'S123321'
+        message_text = get_lvc_notice_text(type='LVC_INITIAL', event_id=self.event_id )
+        self.message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=message_text
+        )
+        GCNLVCNoticeParser().parse(self.message)
```

Comment:
I was looking to here to see how the return value of `parse()` was used. I guess I'm still confused about what the intention is with respect to the Boolean returned.